src/components/input/input.vue

```diff
@@ -0,0 +1,219 @@
+}
+
+function start(el: HTMLElement) {
+  el.style.height = el.scrollHeight + 'px';
+}
+function end(el: HTMLElement) {
+  el.style.height = '';
+}
+
+// dynamic component import
+const AsyncIcon = computed(() => {
+  {
+    const Icon = defineAsyncComponent(
+      () => import('@/components/icons/icons.vue')
+    );
+    return Icon;
+  }
+});
+
+//USE sentContent class for styling in parent
+
+</script>
+
+<template>
+  <div class="input-body">
+    <div :class="[size, 'input-body__top-wrapper', props.disabled ? 'disabled' : '']">
+      <label :for="labelRef" class="body1">
+        {{ props.label }}
+        <div v-if="accordianRef" class="icon-wrapper">
+          <AsyncIcon
+            v-if="accordianRef"
+           
+            class="dropdown"
+            :class="[accordianSwitch ? 'accordian__toggled' : '']"
+            icon="caret-down"
+            @click="toggleAccordian"
+          />
+        </div>
+      </label>
+
+      <transition
+        name="slot"
+        @enter="start"
+        @after-enter="end"
+        @before-leave="start"
+        @after-leave="end"
+      >
+        <slot
+          v-if="accordianSwitch"
+          name="sentContent"
+          class="sent-content"
+          appear
+        />
+      </transition>
+    </div>
+
+    <div
+      :disabled="props.disabled"
+      :class="[
+        size,
+        'input-wrapper',
+        inputClass,
+        props.isValid ? 'valid' : '',
+        props.isValid == false ? 'invalid' : '',
+        props.disabled ? 'disabled' : '',
+      ]"
+    >
+      <input
+        :id="labelRef"
+        autocomplete="off"
+        :type="variantMiddleware"
+        :placeholder="props.placeholder"
+        :disabled="props.disabled"
+        :value="props.modelValue"
+        @input="emitInput($event.target)"
+      >
+
+      <div
+        v-if="variantMiddleware == 'password' || props.variant == 'password'"
+        class="password-controls"
+      >
+        <AsyncIcon
+          v-if="showPassword == false"
+          icon="eye"
+          @click="changePasswordVisibility"
+        />
+        <AsyncIcon
+          v-if="showPassword == true"
+          icon="eye-slash"
+          @click="changePasswordVisibility"
+        />
+      </div>
+      <AsyncIcon
+        v-if="props.isValid == false"
+        class="warning-icon"
+        icon="warning"
+        variant="danger"
+      />
+      <AsyncIcon
+        v-if="props.isValid == true"
+        class="success-icon"
+        icon="check"
+        variant="success"
+      />
+    </div>
+    <div class="user-instructions body2">
+      <p
+        v-if="props.isValid != null"
+        :class="[
+          props.disabled ? 'disabled' : '',
+          props.isValid == true ? 'success-message-text' : '',
+          props.isValid == false ? 'error-message-text' : '',
+          'body2'
+        ]"
+      >
+        {{ props.validationText }}
+      </p>
+      <p class="helper-message-text body2">
```

Comment:
Add v-if="props.helpertext" so the p-element only renders if it has content

Reply:
v-if="props.helpertext != ''" added